File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def cal_limit(schedule):
    plused = schedule % 100 + 10
    new_hour = schedule // 100 + (plused // 60)
    new_min = plused % 60
    return new_hour * 100 + new_min

def solution(schedules, timelogs, startday):
    answer = 0
    
    for i in range(len(schedules)):
        today = startday # 시작 요일
        limit = cal_limit(schedules[i])
        cnt = 0
        for timelog in timelogs[i]:
            if today >= 6:
                today = 1 if today == 7 else today + 1
                continue
            if timelog <= limit:
                cnt += 1
            else:
                break
            today += 1
        if cnt == 5:
            answer += 1
        
    return answer

File: 프로그래머스/1/388351. 유연근무제/유연근무제.java
import java.util.*;

class Solution {
    public int solution(int[] schedules, int[][] timelogs, int startday) {
        int answer = 0;
        
        for (int i = 0; i < schedules.length; i++) {
            int schedule = schedules[i];
            int limit = calcLimit(schedule); // 출근 희망 시각 + 10분 계산
            
            int today = startday;
            int cnt = 0;
            for (int log: timelogs[i]) {
                if (today >= 6) { // 주말이라면
                    today = today % 7 + 1;
                    continue;
                }
                if (log <= limit) {
                    cnt++;
                } else {
                    break;
                }
                today++;
            }
            if (cnt == 5) {
                answer++;
            }
        }
        
        return answer;
    }
    
    private int calcLimit(int schedule) {
        int plused = schedule % 100 + 10;
        int newHour = schedule / 100 + (plused / 60);
        int newMinute = plused % 60;
        return newHour * 100 + newMinute;
    }
}